### ISSUE TITLE
Python SDK compilation on BeagleBone

### ISSUE THREAD

**Issue by Agilart** [Azure/azure-iot-sdks#721]
HI there,
I'm trying to compile the Python SDK directly on my BeagleBone Green, but without success:

c++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.9/README.Bugs> for instructions.
python/src/CMakeFiles/iothub_client_python.dir/build.make:62: recipe for target 'python/src/CMakeFiles/iothub_client_python.dir/iothub_client_python.cpp.o' failed
make[2]: *** [python/src/CMakeFiles/iothub_client_python.dir/iothub_client_python.cpp.o] Error 4
CMakeFiles/Makefile2:2922: recipe for target 'python/src/CMakeFiles/iothub_client_python.dir/all' failed
make[1]: *** [python/src/CMakeFiles/iothub_client_python.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2

I'm using the latest BeagleBone image taken from the site and following the instructions.
I managed to free 431 MB RAM, because there are some posts that this could be the problem.

I managed to compile the C SDK with --skip-unittests. Without this parameter it gives me C++: internal compiler error: Killed (program cc1plus).

Nay ideas how to proceed further. Or if anybody can give me a link to the compiled files.
Thanks in advance

Ivan

**Comment by mregen** [comment]
@Agilart I have seen such errors on Pi2 when swap space is low. Is there a way to increase the swap space similar to Pi2 #458?

**Comment by Agilart** [comment]
@mregen,
thanks for the hint. The BeagleBone does not have Swap by default, so I did:
sudo mkdir -p /var/cache/swap/
sudo dd if=/dev/zero of=/var/cache/swap/swapfile bs=1M count=256
sudo chmod 0600 /var/cache/swap/swapfile
sudo mkswap /var/cache/swap/swapfile
sudo swapon /var/cache/swap/swapfile

This makes a temporary swap file, on reboot it won't be used.

I also stopped all services to free some more mem:
apache2, node-red, cloud9, pulseaudio

Next, I found that nodejs consumes a lot of memory, so I killed it:
ps -A
will list all processes, find the one named nodejs and see the process id

kill 9 <nodejs-process-id>

Finally, I freed all mem cache:
echo 3 > /proc/sys/vm/drop_caches

After all of the above i have:
           total       used       free     shared    buffers     cached
Mem:           492         76        415          0          5         44
-/+ buffers/cache:         26        465
Swap:          255         0        255

Then I ran build and everything went fine, incuding the tests.
I hope this will help other with the same issues.

Best,
Ivan Dragoev
